lib/puppet/provider/package/pip.rb

```diff
@@ -29,7 +29,11 @@ def self.parse(line)
   # that's managed by `pip` or an empty array if `pip` is not available.
   def self.instances
     packages = []
-    pip_cmd = which(cmd) or return []
+    if cmd.class == Array
```

Comment:
I think what @trlinkin may be asking is when will `cmd` ever *not* be an array? I.e. since `cmd` is defined as an array on line 47 below.